matrix_sdk_base/src/models/room.rs

```diff
@@ -614,10 +615,29 @@ impl Room {
     /// Returns true if `MessageQueue` was added to.
     #[cfg(feature = "messages")]
     #[cfg_attr(docsrs, doc(cfg(feature = "messages")))]
-    pub fn handle_message(&mut self, event: &SentMessageEvent) -> bool {
+    pub fn handle_message(&mut self, event: &MessageEventStub<MessageEventContent>) -> bool {
         self.messages.push(event.clone())
     }
 
+    /// Handle a room.redaction event and update the `MessageQueue` if necessary.
+    ///
+    /// Returns true if `MessageQueue` was updated.
+    #[cfg(feature = "messages")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "messages")))]
+    pub fn handle_redaction(&mut self, event: &RedactionEventStub) -> bool {
+        if let Some(msg) = self
+            .messages
+            .iter_mut()
+            .find(|msg| event.redacts == msg.event_id)
+        {
+            // TODO make msg an enum or use AnyMessageEventStub enum to represent
+            msg.content = MessageEventContent::Text(TextMessageEventContent::new_plain("Redacted"));
+            true
+        } else {
+            false
+        }
+    }
```

Comment:
To continue the conversation from the other PR, you want to make `msg` an enum so it can hold either `MessageEventStub<MessageEventContent>` of `RedactionEventStub`?

Reply:
Perhaps just holding the enum we get from the sync is better, people will probably want to restore the timeline as it was, including e.g. memberhip changes.

Reply:
Could we maybe bump up the default amount a `MessageQueue` holds for this then?

Reply:
Sure.